wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -2069,3 +2163,310 @@ instance Example Redemption where
+              case HMS.lookup "message" o of
+                Just "NotEnoughMoney"        ->
+                    NotEnoughMoney
+                        <$> ((o .: "diagnostic") >>= (.: "needMore"))
+                Just "OutputIsRedeem"        ->
+                    OutputIsRedeem <$> ((o .: "diagnostic") >>= (.: "address"))
+                Just "UnknownError"          ->
+                    UnknownError <$> ((o .: "diagnostic") >>= (.: "msg"))
+                Just "InvalidAddressFormat"  ->
+                    InvalidAddressFormat
+                        <$> ((o .: "diagnostic") >>= (.: "msg"))
+                Just "WalletNotFound"        -> pure WalletNotFound
+                Just "WalletAlreadyExists"   ->
+                    WalletAlreadyExists <$> ((o .: "diagnostic") >>= (.: "walletId"))
+                Just "AddressNotFound"       -> pure AddressNotFound
+                Just "TxFailedToStabilize"   -> pure TxFailedToStabilize
+                Just "TxRedemptionDepleted"  -> pure TxRedemptionDepleted
+                Just "TxSafeSignerNotFound"  ->
+                    TxSafeSignerNotFound
+                        <$> ((o .: "diagnostic") >>= (.: "address"))
+                Just "InvalidPublicKey"        ->
+                    InvalidPublicKey <$> ((o .: "diagnostic") >>= (.: "msg"))
+                Just "UnsignedTxCreationError" -> pure UnsignedTxCreationError
+                Just "TooBigTransaction"       -> pure TooBigTransaction
+                Just "SignedTxSubmitError"     ->
+                    SignedTxSubmitError <$> ((o .: "diagnostic") >>= (.: "msg"))
+                Just "CannotCreateAddress"   ->
+                    CannotCreateAddress <$> ((o .: "diagnostic") >>= (.: "msg"))
+                Just "MissingRequiredParams" ->
+                    MissingRequiredParams
+                        <$> ((o .: "diagnostic") >>= (.: "params"))
+                Just _                       ->
+                    fail "Incorrect JSON encoding for WalletError"
+                Nothing                      ->
+                    fail "Incorrect JSON encoding for WalletError"
+        -- WalletIsNotReadyToProcessPayments
+        | HMS.member "estimatedCompletionTime" o = do
+            estCompTO <- (o .: "estimatedCompletionTime")
+            sThroughPO <- (o .: "throughput")
+            prctO <- (o .: "percentage")
+            estCompT <- parseJSON estCompTO
+            sThroughP <- parseJSON sThroughPO
+            prct <- parseJSON prctO
+            return . WalletIsNotReadyToProcessPayments
+                $ SyncProgress estCompT sThroughP prct
+        -- NodeIsStillSyncing
+        | HMS.member "quantity" o = do
+            quantityO <-  o .: "quantity"
+            quantity <- parseJSON quantityO
+            return . NodeIsStillSyncing $ mkSyncPercentage quantity
+        | otherwise = fail "Incorrect JSON encoding for WalletError"
+    parseJSON invalid = typeMismatch "WalletError" invalid
```

Comment:
The only changes compare to the base branch here are:

- Dropped usage of `Pos.Util.Util.aesonError` in favor of `fail` which is doing pretty-much the same thing without 454325 level of indirections.
- Remove the MigrationError & JSONValidationError which are now standalone types.